### PR TITLE
fix: embed cc not shown when seeking back

### LINKED ISSUE #4641: HLS - Embed Closed Captions not shown when seeking back
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes

**What version of Shaka Player are you using?**
4.2.2

**Can you reproduce the issue with our latest release version?**
Yes

**Can you reproduce the issue with the latest code from `main`?**
Yes

**Are you using the demo app or your own custom app?**
Custom app?

**If custom app, can you reproduce the issue using our demo app?**
Yes

**What browser and OS are you using?**
Chrome, Edge in Windows 10

**For embedded devices (smart TVs, etc.), what model and firmware version are you using?**
N/A

**What are the manifest and license server URIs?**

**What configuration are you using?  What is the output of `player.getConfiguration()`?**
{"drm":{"retryParameters":{"maxAttempts":2,"baseDelay":1000,"backoffFactor":2,"fuzzFactor":0.5,"timeout":30000,"stallTimeout":5000,"connectionTimeout":10000},"servers":{},"clearKeys":{},"advanced":{},"delayLicenseRequestUntilPlayed":false,"logLicenseExchange":false,"updateExpirationTime":1,"preferredKeySystems":[],"keySystemsMapping":{},"parseInbandPsshEnabled":false},"manifest":{"retryParameters":{"maxAttempts":3,"baseDelay":1000,"backoffFactor":2,"fuzzFactor":0.5,"timeout":30000,"stallTimeout":5000,"connectionTimeout":10000},"availabilityWindowOverride":100000,"disableAudio":false,"disableVideo":false,"disableText":false,"disableThumbnails":false,"defaultPresentationDelay":0,"segmentRelativeVttTiming":false,"dash":{"clockSyncUri":"","ignoreDrmInfo":false,"disableXlinkProcessing":false,"xlinkFailGracefully":false,"ignoreMinBufferTime":false,"autoCorrectDrift":true,"initialSegmentLimit":1000,"ignoreSuggestedPresentationDelay":false,"ignoreEmptyAdaptationSet":false,"ignoreMaxSegmentDuration":false,"keySystemsByURI":{"urn:uuid:1077efec-c0b2-4d02-ace3-3c1e52e2fb4b":"org.w3.clearkey","urn:uuid:e2719d58-a985-b3c9-781a-b030af78d30e":"org.w3.clearkey","urn:uuid:edef8ba9-79d6-4ace-a3c8-27dcd51d21ed":"com.widevine.alpha","urn:uuid:9a04f079-9840-4286-ab92-e65be0885f95":"com.microsoft.playready","urn:uuid:79f0049a-4098-8642-ab92-e65be0885f95":"com.microsoft.playready","urn:uuid:f239e769-efa3-4850-9c16-a903c6932efb":"com.adobe.primetime"}},"hls":{"ignoreTextStreamFailures":false,"ignoreImageStreamFailures":false,"defaultAudioCodec":"mp4a.40.2","defaultVideoCodec":"avc1.42E01E","ignoreManifestProgramDateTime":false,"mediaPlaylistFullMimeType":"video/mp2t; codecs=\"avc1.42E01E, mp4a.40.2\""}},"streaming":{"retryParameters":{"maxAttempts":2,"baseDelay":1000,"backoffFactor":2,"fuzzFactor":0.5,"timeout":30000,"stallTimeout":5000,"connectionTimeout":10000},"rebufferingGoal":2,"bufferingGoal":10,"bufferBehind":30,"ignoreTextStreamFailures":false,"alwaysStreamText":true,"startAtSegmentBoundary":false,"gapDetectionThreshold":0.5,"durationBackoff":1,"forceTransmuxTS":false,"safeSeekOffset":5,"stallEnabled":true,"stallThreshold":1,"stallSkip":0.1,"useNativeHlsOnSafari":true,"inaccurateManifestTolerance":2,"lowLatencyMode":false,"autoLowLatencyMode":false,"forceHTTPS":false,"preferNativeHls":false,"updateIntervalSeconds":1,"dispatchAllEmsgBoxes":false,"observeQualityChanges":false,"maxDisabledTime":30},"offline":{"usePersistentLicense":true,"numberOfParallelDownloads":5},"abr":{"enabled":false,"useNetworkInformation":true,"defaultBandwidthEstimate":1000000,"switchInterval":8,"bandwidthUpgradeTarget":0.85,"bandwidthDowngradeTarget":0.95,"restrictions":{"minWidth":0,"maxWidth":null,"minHeight":0,"maxHeight":null,"minPixels":0,"maxPixels":null,"minFrameRate":0,"maxFrameRate":null,"minBandwidth":0,"maxBandwidth":null},"advanced":{"minTotalBytes":128000,"minBytes":16000,"fastHalfLife":2,"slowHalfLife":5},"restrictToElementSize":false,"ignoreDevicePixelRatio":false},"preferredAudioLanguage":"","preferredTextLanguage":"eng","preferredVariantRole":"","preferredTextRole":"","preferredAudioChannelCount":2,"preferredVideoCodecs":[],"preferredAudioCodecs":[],"preferForcedSubs":false,"preferredDecodingAttributes":[],"restrictions":{"minWidth":0,"maxWidth":null,"minHeight":0,"maxHeight":null,"minPixels":0,"maxPixels":null,"minFrameRate":0,"maxFrameRate":null,"minBandwidth":0,"maxBandwidth":null},"playRangeStart":0,"playRangeEnd":null,"cmcd":{"enabled":false,"sessionId":"","contentId":"","useHeaders":false}}

**What did you do?**
Using an HLS stream with CEA-X08 captions, select this from the captions list and seek forward and backward.

**What did you expect to happen?**
See the captions.

**What actually happened?**
If you seek forward yo see the captions, but if you then seek backward, the captions are not shown.

### PR DESCRIPTION
**Pull request by g-meola**
For HLS non native, Shaka gets the captions from mux.js transmuxer. This
 works until you seek back before the buffered range. In mux.js there is
 a condition that ignores a caption if this belongs to a DTS that is
behind the latest it already has parsed.

To handle seeks, it adds a reset method that is called by the transmuxer.resetCaptions() method in mp4\transmuxer.js in mux.js. But Shaka Player is not calling it. So I added the resetCaptions() method to the mux.js extern definition and we called it in the appendBuffer method in media_source_engine.js before sending the data to transmux and only if the seeked parameter is True.

Closes #4641

### CODE REVIEW

File: externs/mux.js [hunk 1]
```diff
@@ -167,4 +170,3 @@ muxjs.mp4.ParsedClosedCaptions;
  * @property {string} text The content of the closed caption.
  */
 muxjs.mp4.ClosedCaption;
-
```
Comment (avelad) [inline externs/mux.js:170]: Please revert this change

File: test/media/media_source_engine_integration.js [hunk 2]
```diff
@@ -388,6 +400,25 @@ describe('MediaSourceEngine', () => {
...
+    // Load TS file with CEA-708 captions.
+    metadata = shaka.test.TestScheme.DATA['cea-708_ts'];
+    generators = shaka.test.TestScheme.GENERATORS['cea-708_ts'];
+
+    const initObject = new Map();
+    initObject.set(ContentType.VIDEO, getFakeStream(metadata.video));
+    initObject.set(ContentType.TEXT, getFakeStream(metadata.text));
+    // Call with forceTransmuxTS = true, so that it will transmux even on
+    // platforms with native TS support.
+    await mediaSourceEngine.init(initObject, /* forceTransmuxTS= */ true);
```
Comment (avelad) [inline test/media/media_source_engine_integration.js:413]: The name of `forceTransmuxTS` has been changed, it is now called `forceTransmux`, please change it.

File: test/media/media_source_engine_integration.js [hunk 3]
```diff
@@ -388,6 +400,25 @@ describe('MediaSourceEngine', () => {
...
 
+  it('extracts CEA-708 captions from previous segment from hls', async () => {
+    // Load TS file with CEA-708 captions.
+    metadata = shaka.test.TestScheme.DATA['cea-708_ts'];
+    generators = shaka.test.TestScheme.GENERATORS['cea-708_ts'];
+
+    const initObject = new Map();
+    initObject.set(ContentType.VIDEO, getFakeStream(metadata.video));
+    initObject.set(ContentType.TEXT, getFakeStream(metadata.text));
+    // Call with forceTransmuxTS = true, so that it will transmux even on
```
Comment (avelad) [inline test/media/media_source_engine_integration.js:411]: The name of `forceTransmuxTS` has been changed, it is now called `forceTransmux`, please change it.

Comment (avelad) [inline test/media/media_source_engine_integration.js:411]: The name of `forceTransmuxTS` has been changed, it is now called `forceTransmux`, please change it.

**Review by avelad — CHANGES_REQUESTED** [review]
Please, review my previous comments.

File: test/media/media_source_engine_integration.js (hunk 3 above)
Comment (g-meola) [reply test/media/media_source_engine_integration.js]: Missed that. Done. Thanks

Comment (g-meola) [reply test/media/media_source_engine_integration.js]: Missed that. Done. Thanks

File: test/media/media_source_engine_integration.js (hunk 2 above)
Comment (g-meola) [reply test/media/media_source_engine_integration.js]: Missed that. Done. Thanks

File: lib/media/transmuxer.js [hunk 4]
```diff
@@ -255,4 +261,3 @@ shaka.media.Transmuxer = class {
     this.isTransmuxing_ = false;
   }
 };
-
```
Comment (avelad) [inline lib/media/transmuxer.js:258]: please revert this change

**Review by avelad — CHANGES_REQUESTED** [review]
LGTM, only a small change it's required

File: lib/media/transmuxer.js (hunk 4 above)
Comment (g-meola) [reply lib/media/transmuxer.js]: Done. Thanks!

**Review by avelad — APPROVED** [review]

**Review by theodab — APPROVED** [review]

**Pull request merged**